### PR TITLE
Add more unit test for SelectStatementContext

### LINKED ISSUE #14672: Add more unit test for SelectStatementContext
Hi, community:

This issue is to add more unit tests for the SelectStatementContext class. Welcome to claim them.

The specific subtasks are as follows:

- [ ]  Add unit test for SelectStatementContext#isContainsDollarParameterMarker
- [ ]  Add unit test for SelectStatementContext#isContainsPartialDistinctAggregation

Through this issue, you can understand the process of participating in Apache ShardingSphere, and be familiar with sharding.

### PR DESCRIPTION
**Pull request by Bpazy**
Fixes #14672.

Changes proposed in this pull request:
- Add unit test for SelectStatementContext#isContainsDollarParameterMarker
- Add unit test for SelectStatementContext#isContainsPartialDistinctAggregation

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java [hunk 1]
```diff
@@ -484,7 +489,85 @@ private void assertContainsSubqueryWhereEmpty(final SelectStatement selectStatem
...
+    }
+    
+    private void assertContainsDollarParameterMarker(final SelectStatement selectStatement) {
+        ProjectionsSegment projectionsSegment = mock(ProjectionsSegment.class);
+        when(projectionsSegment.getProjections()).thenReturn(Lists.newArrayList(new ParameterMarkerExpressionSegment(0, 0, 0, ParameterMarkerType.DOLLAR)));
+        selectStatement.setProjections(projectionsSegment);
+        SelectStatementContext selectStatementContext = new SelectStatementContext(Collections.singletonMap(DefaultDatabase.LOGIC_NAME, mock(ShardingSphereDatabase.class)), Collections.emptyList(),
+                selectStatement, DefaultDatabase.LOGIC_NAME);
+        assertTrue(selectStatementContext.isContainsDollarParameterMarker());
+
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java:525]: Please remove this useless blank line.

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java [hunk 2]
```diff
@@ -484,7 +489,85 @@ private void assertContainsSubqueryWhereEmpty(final SelectStatement selectStatem
...
+    }
+    
+    @Test
+    public void assertContainsDollarParameterMarkerForSQLServer() {
+        assertContainsDollarParameterMarker(new SQLServerSelectStatement());
+    }
+    
+    private void assertContainsDollarParameterMarker(final SelectStatement selectStatement) {
+        ProjectionsSegment projectionsSegment = mock(ProjectionsSegment.class);
+        when(projectionsSegment.getProjections()).thenReturn(Lists.newArrayList(new ParameterMarkerExpressionSegment(0, 0, 0, ParameterMarkerType.DOLLAR)));
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java:520]: Please do not use guava Lists.

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java [hunk 3]
```diff
@@ -484,7 +489,85 @@ private void assertContainsSubqueryWhereEmpty(final SelectStatement selectStatem
...
+    public void assertContainsPartialDistinctAggregationForSQLServer() {
+        assertContainsPartialDistinctAggregation(new SQLServerSelectStatement());
+    }
+
+    private void assertContainsPartialDistinctAggregation(final SelectStatement selectStatement) {
+        ProjectionsSegment projectionsSegment = new ProjectionsSegment(0, 0);
+        projectionsSegment.getProjections().add(new AggregationProjectionSegment(0, 0, AggregationType.COUNT, "(*)"));
+        projectionsSegment.getProjections().add(new AggregationDistinctProjectionSegment(0, 10, AggregationType.COUNT, "(1)", "distinctExpression"));
+        selectStatement.setProjections(projectionsSegment);
+
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java:565]: Please remove this useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java (hunk 3 above)
Comment (Bpazy) [reply shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java]: Done

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java (hunk 2 above)
Comment (Bpazy) [reply shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java]: Done

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java (hunk 1 above)
Comment (Bpazy) [reply shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/impl/SelectStatementContextTest.java]: Done

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**